### PR TITLE
Implemented unjoin.

### PR DESCRIPTION
**Pull request by gyliu513**
Fixed https://github.com/kubernetes-sigs/federation-v2/issues/127

/cc @marun

### CODE REVIEW

**Comment by gyliu513** [comment]
```
LiuGuangyas-MacBook-Pro:federation-v2 gyliu$ bin/kubefnord --help
kubefnord controls a Kubernetes Cluster Federation. Find more information at https://github.com/kubernetes-sigs/federation-v2.

Usage:
  kubefnord [flags]
  kubefnord [command]

Available Commands:
  join        Join a cluster to a federation
  unjoin      Unjoin a cluster from a federation

Flags:
      --alsologtostderr                  log to standard error as well as files
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-flush-frequency duration     Maximum number of seconds between log flushes (default 5s)
      --logtostderr                      log to standard error instead of files (default true)
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          log level for V logs
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging

Use "kubefnord [command] --help" for more information about a command.
```

File: pkg/kubefnord/unjoin.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+	}
+
+	glog.V(2).Info("Deleted role binding for service account in unjoining cluster")
+
+	glog.V(2).Info("Deleting service account in unjoining cluster")
+
+	err = deleteServiceAccount(unjoiningClusterClientset, namespace,
+		unjoiningClusterName, hostClusterName, dryRun)
+	if err != nil {
+		glog.V(2).Infof("Error creating service account in joining cluster: %v", err)
```
Comment (onyiny-ang) [inline pkg/kubefnord/unjoin.go]: s/creating/deleting  s/joining/unjoining

File: pkg/kubefnord/unjoin.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+	if err != nil {
+		// TODO(font): Return new error with this same text so it can be output
+		// by caller.
+		glog.V(2).Infof("Failed to get host cluster config: %v", err)
+		return err
+	}
+
+	clusterConfig, err := config.ClusterConfig(j.clusterContext, j.Kubeconfig)
+	if err != nil {
+		glog.V(2).Infof("Failed to get joining cluster config: %v", err)
```
Comment (onyiny-ang) [inline pkg/kubefnord/unjoin.go]: s/joining/unjoining
I'm not sure that using "unjoining" in this way is ideal. It is consistent with the join command though! 
@marun, thoughts?

File: pkg/kubefnord/unjoin.go [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+	_, err = deleteFederatedCluster(clusterClientset, fedClientset, federationNamespace,
+		unjoiningClusterName, dryRun)
+	if err != nil {
+		glog.V(2).Infof("Failed to create federated cluster resource: %v", err)
+		return err
+	}
+
+	glog.V(2).Info("Deleted federated cluster resource")
+
+	glog.V(2).Infof("Creating %s namespace in unjoining cluster", federationNamespace)
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply pkg/kubefnord/unjoin.go]: Done, deleted namespace here.

File: pkg/kubefnord/unjoin.go [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+func removeFromClusterRegistry(hostConfig *rest.Config, host, unjoiningClusterName string,
+	dryRun bool) error {
+	// Get the cluster registry clientset using the host cluster config.
+	crClientset, err := util.ClusterRegistryClientset(hostConfig)
+	if err != nil {
+		glog.V(2).Infof("Failed to get cluster registry clientset: %v", err)
+		return err
+	}
+
+	glog.V(2).Info("UnRegistering cluster with the cluster registry.")
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply pkg/kubefnord/unjoin.go]: Done

File: pkg/kubefnord/unjoin.go [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+
+	err = unRegisterCluster(crClientset, host, unjoiningClusterName, dryRun)
+	if err != nil {
+		glog.V(2).Infof("Could not unregister cluster with the cluster registry: %v", err)
+		return err
+	}
+
+	glog.V(2).Info("UnRegistered cluster with the cluster registry.")
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply pkg/kubefnord/unjoin.go]: Done

File: pkg/kubefnord/unjoin.go [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+	return nil
+}
+
+// unRegisterCluster unregisters a cluster with the cluster registry.
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply pkg/kubefnord/unjoin.go]: Done

File: pkg/kubefnord/unjoin.go (hunk 1 above)
Comment (gyliu513) [reply pkg/kubefnord/unjoin.go]: This function is not needed

File: pkg/kubefnord/unjoin.go (hunk 2 above)
Comment (onyiny-ang) [reply pkg/kubefnord/unjoin.go]: sgtm

**Comment by gyliu513** [comment]
This was replaced by https://github.com/kubernetes-sigs/federation-v2/pull/137

**Pull request closed without merging**